optaplanner-core/src/main/java/org/optaplanner/core/impl/partitionedsearch/DefaultPartitionedSearchPhase.java

```diff
@@ -48,10 +50,10 @@
  * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
  */
 public class DefaultPartitionedSearchPhase<Solution_> extends AbstractPhase<Solution_>
-        implements PartitionedSearchPhase<Solution_>, PartitionedSearchPhaseLifecycleListener<Solution_> {
+        implements PartitionedSearchPhase<Solution_> {
 
     protected SolutionPartitioner<Solution_> solutionPartitioner;
-    protected ThreadPoolExecutor threadPoolExecutor;
+    protected ThreadFactory threadFactory;
```

Comment:
The phase needs to have the thread factory if `solver.solve()` should be repeatable. At the end of the phase, thread pool is shut down and so a new thread pool has to be created at the phase start.